control-plane/p4RuntimeArchStandard.h

```diff
@@ -916,6 +924,139 @@ class P4RuntimeArchHandlerCommon : public P4RuntimeArchHandlerIface {
+        auto sym_id = symbols.getId(sym_type, decl->controlPlaneName());
+        uint32_t entries_arg_idx = UINT32_MAX;
+        uint32_t default_arg_idx = UINT32_MAX;
+        for (uint32_t idx = 0; idx < decl->arguments->size(); idx++) {
+            auto arg_name = decl->arguments->at(idx)->name.originalName;
+            if (arg_name == ENTRIES_ARG_NAME)
+                entries_arg_idx = idx;
+            else if (arg_name == DEFAULT_ARG_NAME)
+                default_arg_idx = idx;
+        }
+        // handle const_entries
+        if (entries_arg_idx != UINT32_MAX) {
+            const IR::Argument* arg = decl->arguments->at(entries_arg_idx);
+            auto arg_name = arg->name.originalName;
+            if (arg->expression->is<IR::ListExpression>() == false) {
+                ::error(ErrorType::ERR_UNEXPECTED, "Unexpected expression for %1% in %2%, "
+                        "expected ListExpression", arg_name, decl);
+                return;
+            }
+            uint32_t e_cnt = 1;
+            for (auto e : arg->expression->to<IR::ListExpression>()->components) {
+                if (e->is<IR::ListExpression>() == false) {
+                    ::error(ErrorType::ERR_UNEXPECTED, "Unexpected expression for entry "
+                            "%1% of %2%", e_cnt, arg_name);
+                    return;
+                }
+                keyBitwidth =
+                    decl->type->to<IR::Type_Specialized>()->arguments->at(0)->width_bits();
+                add_mvlut_entry(e->to<IR::ListExpression>(), mvlut_type_name, sym_id, false);
+            }
+        }
+
+        // handle default_value
+        if (default_arg_idx != UINT32_MAX) {
+            const IR::Argument* arg = decl->arguments->at(default_arg_idx);
+            add_mvlut_entry(arg->expression, mvlut_type_name, sym_id, true);
+        }
+    }
+    void add_mvlut_entry(const IR::Expression* entry_exp,
+                                               cstring mvlut_type_name,
+                                               p4rt_id_t rt_symbol_id,
+                                               bool is_default) {
+        CHECK_NULL(this->rtEntries);
+        auto protoUpdate = const_cast<p4::v1::WriteRequest*>(rtEntries)->add_updates();
+        protoUpdate->set_type(p4::v1::Update::INSERT);
+        auto protoEntity = protoUpdate->mutable_entity();
+        auto protoEntry = protoEntity->mutable_extern_entry();
+        protoEntry->set_extern_type_id((rt_symbol_id & 0xff000000) >> 24);
+        // create an MVLUT proto object and set fields
+        p4::v1::MVLUTEntry mvlut_entry;
+        mvlut_entry.set_mvlut_id(rt_symbol_id);
+        // add 0 as key expr for default_value
+        uint32_t paramId = 1;
+        if (is_default == true) {
+            add_mvlut_match_entry(&mvlut_entry, mvlut_type_name, new IR::Constant(0));
+        }
+        if (entry_exp->is<IR::ListExpression>()) {
```

Comment:
number of values in key place and fields list are not validated by frontend as the prototype allows don't care type for const_entries. We may need to do semantic check in the backend.